### PR TITLE
fix(pkce): match pkce implementation to specifications

### PR DESCRIPTION
**Pull request by vladikoff**
Fixes https://github.com/mozilla/fxa-oauth-server/issues/495

### CODE REVIEW

**Comment by GitCop** [comment]
Thanks for contributing! Unfortunately, I'm here to tell you there were the following style issues with your Pull Request:

 * Commit: 8a77576484b076f0a21140907c73b0b499ae2a40
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/mozilla/fxa-oauth-server

---
This message was auto-generated by https://gitcop.com

**Comment by vladikoff** [comment]
@rfk r?

File: test/routes/token.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ function joiNotAllowed(err, param) {
   assert.equal(err.details[0].message, `"${param}" is not allowed`);
 }
 
-describe('/token POST', function () {
+describe.only('/token POST', function () {
```
Comment (rfk) [inline test/routes/token.js]: Actually, what's this `only`, is it a testing leftover?

Comment (vladikoff) [reply test/routes/token.js]: > is it a testing leftover? 

ah yes removing..

**Comment by GitCop** [comment]
Thanks for contributing! Unfortunately, I'm here to tell you there were the following style issues with your Pull Request:

 * Commit: 8a77576484b076f0a21140907c73b0b499ae2a40
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/mozilla/fxa-oauth-server

---
This message was auto-generated by https://gitcop.com

**Pull request merged**